### PR TITLE
Add mainnet fork deployment tests to CI, extend the network test to perform various health checks, and set a periodic task (every 5 minutes) to run these checks

### PR DESCRIPTION
**Pull request by lbeder**

### CODE REVIEW

File: deploy/tests/network.ts [hunk 1]
```diff
@@ -77,76 +81,320 @@ describe('network', () => {
...
+                const prevVBNTTokenAmount = await getBalance(vbnt, bntWhale);
+                const prevTotalSupply = await bnt.totalSupply();
+
+                await bnt.connect(bntWhale).approve(network.address, bntAmount);
+                await network.connect(bntWhale).deposit(bnt.address, bntAmount);
+
+                const receivedBNBNTAmount = (await getBalance(bntBNT, bntWhale)).sub(prevBNBNTAmount);
+
+                expect(receivedBNBNTAmount).be.gt(0);
+                expect(await getBalance(vbnt, bntWhale)).to.equal(prevVBNTTokenAmount.add(receivedBNBNTAmount));
```
Comment (barakman) [inline deploy/tests/network.ts:260]: I'd use:
```
const currBNBNTAmount = await getBalance(bntBNT, bntWhale);

expect(currBNBNTAmount).be.gt(prevBNBNTAmount);
expect(await getBalance(vbnt, bntWhale)).to.equal(prevVBNTTokenAmount.add(currBNBNTAmount).sub(prevBNBNTAmount));
```

**Review by barakman — APPROVED** [review]

File: deploy/tests/network.ts [hunk 2]
```diff
@@ -77,76 +81,320 @@ describe('network', () => {
...
+            ]);
+
+            await expectRoleMembers(externalRewardsVault, Roles.Upgradeable.ROLE_ADMIN, [daoMultisig]);
+            await expectRoleMembers(externalRewardsVault, Roles.Vault.ROLE_ASSET_MANAGER, [standardRewards.address]);
+
+            await expectRoleMembers(poolTokenFactory, Roles.Upgradeable.ROLE_ADMIN, [daoMultisig]);
+
+            await expectRoleMembers(networkSettings, Roles.Upgradeable.ROLE_ADMIN, [daoMultisig]);
+
+            await expectRoleMembers(bntPool, Roles.Upgradeable.ROLE_ADMIN, [daoMultisig, network.address]);
```
Comment (yudilevi) [inline deploy/tests/network.ts:145]: why does the network need the admin role of the BNT pool?

File: deploy/tests/network.ts [hunk 3]
```diff
@@ -77,76 +81,320 @@ describe('network', () => {
...
+            await expectRoleMembers(bntPool, Roles.BNTPool.ROLE_BNT_POOL_TOKEN_MANAGER);
```
Comment (yudilevi) [inline deploy/tests/network.ts:146]: will be the auto compounding rewards, right?

File: deploy/tests/network.ts (hunk 2 above)
Comment (lbeder) [reply deploy/tests/network.ts:145]: It needs it to grant/revoke roles when processing the addition/removal of pool collections

File: deploy/tests/network.ts (hunk 3 above)
Comment (lbeder) [reply deploy/tests/network.ts:146]: Yeah (see here https://github.com/bancorprotocol/contracts-v3/pull/339/files#diff-51e7f998be6b9092c56b78c26d1d65efeb279ad11f2b6b7a16c6219d077f4082)

File: deploy/tests/network.ts (hunk 2 above)
Comment (yudilevi) [reply deploy/tests/network.ts:145]: oh right 👍

File: deploy/tests/network.ts (hunk 3 above)
Comment (yudilevi) [reply deploy/tests/network.ts:146]: cool 👍

**Review by yudilevi — APPROVED** [review]

File: deploy/tests/network.ts (hunk 1 above)
Comment (lbeder) [reply deploy/tests/network.ts:260]: The current check is explicit and I don't see how your suggestion improves on it. Moving forward.

**Pull request merged**